domain/modeldefaults/state/state.go

```diff
@@ -287,63 +359,142 @@ WHERE m.uuid = $modelUUID.uuid
-// If the cloud is not found, an error specifying [clouderrors.NotFound] is returned.
+// If the cloud is not found, an error specifying [clouderrors.NotFound] is
+// returned.
 func (s *State) GetCloudUUID(ctx context.Context, cloudName string) (cloud.UUID, error) {
 	db, err := s.DB()
 	if err != nil {
-		return "", errors.Trace(err)
+		return "", errors.Capture(err)
 	}
-	cld := dbCloud{Name: cloudName}
+	cloudNameVal := cloudNameValue{Name: cloudName}
 	query, err := s.Prepare(`
-SELECT &dbCloud.uuid
+SELECT &cloudUUIDValue.*
 FROM cloud c
-WHERE c.name = $dbCloud.name
-`, cld)
+WHERE c.name = $cloudNameValue.name
+`, cloudNameVal, cloudUUIDValue{})
 	if err != nil {
-		return "", errors.Trace(err)
+		return "", errors.Capture(err)
 	}
 
+	cloudUUIDVal := cloudUUIDValue{}
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		err := tx.Query(ctx, query, cld).Get(&cld)
+		err := tx.Query(ctx, query, cloudNameVal).Get(&cloudUUIDVal)
 		if errors.Is(err, sqlair.ErrNoRows) {
-			return interrors.Errorf("cloud %q not found", cloudName).Add(clouderrors.NotFound)
+			return clouderrors.NotFound
 		}
-		return errors.Trace(err)
+		return err
+	})
+
+	if err != nil {
+		return cloud.UUID(""), errors.Errorf(
+			"getting cloud UUID for cloud with name %q: %w", cloudName, err,
+		)
+	}
+
+	return cloud.UUID(cloudUUIDVal.UUID), nil
+}
+
+// SetCloudDefaults is responsible for removing any previously set cloud
+// default values and setting the new cloud defaults to use. If no defaults are
+// supplied to this function then the currently set cloud default values will be
+// removed and no further operations will be performed. If no cloud exists for
+// the cloud name then an error satisfying [clouderrors.NotFound] is returned.
+func SetCloudDefaults(
+	ctx context.Context,
+	tx *sqlair.TX,
+	cloudName string,
+	defaults map[string]string,
+) error {
+	cloudNameVal := cloudNameValue{Name: cloudName}
+	var cloudUUIDVal cloudUUIDValue
+	cloudUUIDStmt, err := sqlair.Prepare(`
+SELECT &cloudUUIDValue.*
+FROM cloud 
+WHERE name = $cloudNameValue.name
+`, cloudNameVal, cloudUUIDVal)
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	err = tx.Query(ctx, cloudUUIDStmt, cloudNameVal).Get(&cloudUUIDVal)
+	if errors.Is(err, sqlair.ErrNoRows) {
+		return errors.Errorf(
+			"setting cloud %q defaults, cloud does not exist", cloudName,
+		).Add(clouderrors.NotFound)
+	} else if err != nil {
+		return errors.Errorf(
+			"getting cloud %q uuid to set cloud defaults: %w", cloudName, err,
+		)
+	}
+
+	deleteStmt, err := sqlair.Prepare(`
+DELETE FROM cloud_defaults 
+WHERE       cloud_defaults.cloud_uuid = $cloudUUIDValue.uuid
+`, cloudUUIDVal)
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	err = tx.Query(ctx, deleteStmt, cloudUUIDVal).Run()
+	if err != nil {
+		return errors.Errorf("removing previously set cloud %q defaults: %w", cloudName, err)
+	}
+
+	if len(defaults) == 0 {
```

Comment:
Just want to confirm, do we want to delete all if the defaults arg is empty instead of a no-ops?

Reply:
yep this is correct behaviour. Set is an explicit operation to say make this resemble x